src/osd/mac/video.cpp

```diff
@@ -108,15 +108,23 @@ void mac_osd_interface::update(bool skip_redraw)
 //      profiler_mark(PROFILER_END);
 	}
 
-	// poll the joystick values here
-	downcast<mac_osd_interface&>(machine().osd()).poll_inputs(machine());
-
-	check_osd_inputs(machine());
 	// if we're running, disable some parts of the debugger
 	if ((machine().debug_flags & DEBUG_FLAG_OSD_ENABLED) != 0)
 		debugger_update();
 }
 
+//============================================================
+//  input_update
+//============================================================
+
+void mac_osd_interface::input_update(void)
+{
+	// poll the joystick values here
+	downcast<mac_osd_interface&>(machine().osd()).process_events_buf();
+	downcast<mac_osd_interface&>(machine().osd()).poll_inputs(machine());
```

Comment:
Why are you calling `machine().osd()` and doing a downcast here when you’re already in a `mac_osd_interface` member?